lib/rules/padding-lines-in-component-definition.js

```diff
@@ -0,0 +1,350 @@
+      const commaToken = sourceCode.getTokenAfter(currentProperty, isComma)
+
+      const start = commaToken ? commaToken.range[1] : currentProperty.range[1]
+      const end = nextProperty.range[0]
+
+      const paddingText = sourceCode.text.slice(start, end)
+      const newText = `\n${splitLines(paddingText).pop()}`
+
+      return fixer.replaceTextRange([start, end], newText)
+    }
+
+    /**
+     * @param {Property} currentProperty
+     * @param {Property} nextProperty
+     * @param {RuleFixer} fixer
+     * @param {number} betweenLinesRange
+     * */
+    function insertLines(
+      currentProperty,
+      nextProperty,
+      fixer,
+      betweenLinesRange
+    ) {
+      const commaToken = sourceCode.getTokenAfter(currentProperty, isComma)
+
+      const lineBeforeNextProperty =
+        sourceCode.lines[nextProperty.loc.start.line - 1]
+      const lastSpaces = /** @type {RegExpExecArray} */ (
+        /^\s*/.exec(lineBeforeNextProperty)
+      )[0]
+
+      const newText = betweenLinesRange === 0 ? `\n\n${lastSpaces}` : '\n'
+      return fixer.insertTextAfter(commaToken || currentProperty, newText)
+    }
+
+    /**
+     * @param {Property[]} properties
+     * @param {any} option
+     * @param {any} nextOption
+     * */
+    function verify(properties, option, nextOption) {
+      const groupSingleLineProperties = parseBooleanOption(
+        options,
+        OptionKeys.GroupSingleLineProperties
+      )
+
+      for (const [i, currentProperty] of properties.entries()) {
+        const nextProperty = properties[i + 1]
+
+        if (nextProperty && option !== AvailablePaddingOptions.Ignore) {
+          const betweenLinesRange =
+            nextProperty.loc.start.line - currentProperty.loc.end.line
+
+          if (
+            needGroupSingleLineProperties(
+              currentProperty,
+              nextProperty,
+              groupSingleLineProperties
+            )
+          ) {
+            if (betweenLinesRange > 1) {
+              context.report({
+                node: nextProperty.key,
+                messageId: 'groupSingleLineProperties',
+                loc: nextProperty.loc,
+                fix(fixer) {
+                  return replaceLines(currentProperty, nextProperty, fixer)
+                }
+              })
+            }
+            continue
+          }
+
+          if (
+            betweenLinesRange <= 1 &&
+            option === AvailablePaddingOptions.Always
+          ) {
+            context.report({
+              node: nextProperty.key,
+              messageId: 'always',
+              loc: nextProperty.loc,
+              fix(fixer) {
+                return insertLines(
+                  currentProperty,
+                  nextProperty,
+                  fixer,
+                  betweenLinesRange
+                )
+              }
+            })
+          } else if (
+            betweenLinesRange > 1 &&
```

Comment:
I think we should support cases where comments and non property node are included. Same for `groupSingleLineProperties`.

In the current implementation, using 'never' can cause comments and nodes to be lost due to auto-fix.

```vue
<script >
/* eslint vue/padding-lines-in-component-definition: ["error", "never"] */
export default {
  props: {
    a:{},
    /** lost */
    b:{}
  },
}
</script>
```

```vue
<script>
/* eslint vue/padding-lines-in-component-definition: ["error", "never"] */
export default {
  a: {},
  ...lost,
  b: {},
}
</script>
```

Reply:
@ota-meshi Thank you for review! I have a question about comments. If we have comments, should we implement them as part of property or as own property? As an example, what is the correct variant?

```vue
<script>
/* eslint vue/padding-lines-in-component-definition: ["error", "always"] */
export default {
  props: {
    a:{},
    
    /** lost */
    
    b:{}
  },
}
</script>
```
or
```vue
<script>
/* eslint vue/padding-lines-in-component-definition: ["error", "always"] */
export default {
  props: {
    a:{},
    
    /** lost */
    b:{}
  },
}
</script>
```

Also, will the behavior differ if we use JSDoc or multiline comments? 

Reply:
I think it should be styled like this: 

```js
export default {
  props: {
    a:{},
    
    /** lost */
    b:{}
  },
}
```

I think any form of comment should behave the same. It can also be a suppressing comment.

```js
export default {
  props: {
    a:{},
    
    // eslint-disable-next-line rule
    b:{}
  },
}
```

Reply:
Thanks for quick response! 🙏
I added comments and non-property node support